### PR TITLE
Fix a couple places where utf-8 was not specified explicitly for Path.read_text/write_text

### LINKED ISSUE #1977: Getting an error when use the cci error gist command
**Describe the bug**
I am getting an error when using the `cci error gist` command

**Reproduction steps**
1. I happened to be in the sal repo
2. I was creating a prerelease org using the regression_org flow
3. Once it completed I used the "cci error gist" command
4. I got an error:
Error: 'charmap' codec can't decode byte 0x90 in position 106756: character maps to <undefined>
(Similar to https://github.com/SFDO-Tooling/CumulusCI/issues/1619)

**Context**
I wanted to review a flow's output so I used the cci error gist command

CumulusCI version: 3.17.1 (C:\Users\anne.jackson\.local\bin\cci)
Python version: 3.8.2 (c:\users\anne.jackson\.local\pipx\venvs\cumulusci\scripts\python.exe)
- Windows 10
- I am using Windows Visual Studio Code Terminal

### PR DESCRIPTION
**Pull request by davisagli**
Without this it would use locale.getpreferredencoding(), which at least in some cases is cp1252 in Windows.

Also includes a fix for #1514 to make `cci error info` show the full traceback by default.

# Critical Changes

# Changes
- The `cci error info` command now defaults to showing the entire traceback when it is more than 30 lines.

# Issues Closed
- Fixed a bug where `cci error gist` could throw a UnicodeDecodeError (fixes #1977)

### CODE REVIEW

File: cumulusci/cli/cci.py [hunk 1]
```diff
@@ -1590,12 +1592,11 @@ def lines_from_traceback(log_content, num_lines):
         return f"\nNo stacktrace found in: {CCI_LOGFILE_PATH}\n"
 
     stacktrace = ""
-    for line in reversed(log_content.split("\n")):
+    for i, line in enumerate(reversed(log_content.split("\n"))):
```
Comment (prescod) [inline cumulusci/cli/cci.py:1595]: Minor: If you use enumerate(..., 1) then you won't have to add 1 later.

File: cumulusci/cli/cci.py [hunk 2]
```diff
@@ -1590,12 +1592,11 @@ def lines_from_traceback(log_content, num_lines):
...
         stacktrace = "\n" + line + stacktrace
         if stacktrace_start in line:
             break
-        num_lines -= 1
-        if num_lines == -1:
+        if max_lines is not None and i + 1 == max_lines:
```
Comment (prescod) [inline cumulusci/cli/cci.py:1599]: Minor: if you let 0 mean "unlimited" then it simplifies the code.

**Review by prescod — DISMISSED** [review]
If you are feeling ambitious you could add a test for re-raised exception syntax.

**Comment by davisagli** [comment]
@jstvz I found https://pypi.org/project/flake8-file-encoding/ but it only checks `open`, not the `Path` methods.

**Comment by annejackson** [comment]
Windows is such a trouble-maker!!

**Pull request merged**